library/src/main/java/net/grandcentrix/tray/core/Preferences.java

```diff
@@ -301,4 +302,11 @@ static boolean isDataTypeSupported(final Object data) {
                 || data instanceof Boolean
                 || data == null;
     }
+
+    boolean putData(String key, Object value) {
```

Comment:
If it should be accessible from any other class like subclasses, than I am missing the check if the version was upgraded and an appropriate logging.
I would just make it `private`.